Sources/Request/Call.swift

```diff
@@ -0,0 +1,123 @@
+public enum HTTPMethod: String {
+	case GET, POST, PUT, DELETE, PATCH
+}
+
+
+
+/// Defines a request that will be called in the Service
+/// You can add `[Parameter]` to the request and optionally authenticate the request when needed.
+open class Call {
+	open let path: String
+	open let httpMethod: HTTPMethod
+	open var rootNode: String?
+	open var parameter: [Parameter]?
+	open var authenticate: ((_ request: inout URLRequest) -> Void)?
+
+	/// Initializes Call to retreive object(s) from the server.
+	/// parameter path: the path to point the call too
+	/// parameter method: the method to use for the urlRequest
+	/// parameter rootNode: used to extract JSON in method `rootNode(from:)`.
+	/// parameter serializableModel: the model is put into the body of the request as json.
+	/// parameter authenticate: optionally add authentication information to the request. Every time a request the authenticate function is called. So you can deal with authentication methods that change over time
+	public convenience init<T: Serializable> (path: String, method: HTTPMethod = .POST, rootNode: String? = nil, serializableModel: T, authenticate: ((_ request: inout URLRequest) -> Void)? = nil) {
+		self.init(path: path, method: method, rootNode: rootNode, parameter: [.jsonNode(serializableModel.json)], authenticate: authenticate)
+	}
+
+	/// Initializes Call to retreive object(s) from the server.
+	/// parameter path: the path to point the call too
+	/// parameter method: the method to use for the urlRequest
+	/// parameter rootNode: used to extract JSON in method `rootNode(from:)`.
+	/// parameter parameter: array of parameters to be added to the request when created.
+	/// parameter authenticate: optionally add authentication information to the request. Every time a request the authenticate function is called. So you can deal with authentication methods that change over time
+	public init(path: String, method: HTTPMethod = .GET, rootNode: String? = nil, parameter: [Parameter]? = nil, authenticate: ((_ request: inout URLRequest) -> Void)? = nil) {
+		self.path = path
+		self.httpMethod = method
+		self.rootNode = rootNode
+		self.parameter = parameter
+		self.authenticate = authenticate
+	}
+
+	open func request(withConfiguration configuration: Configuration) -> URLRequest? {
+		var request = URLRequest(url: URL(string: "\(configuration.baseURL)/\(path)")!, cachePolicy: .reloadIgnoringLocalAndRemoteCacheData) // uses default timeout
+		request.httpMethod = httpMethod.rawValue
+		request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+		insertParameter(request: &request)
+		authenticate?(&request)
```

Comment:
Lukte het niet met een protocol (eg `Authorizeable`), dat eg een `var headers:[String:String]`? Ik zou denken dat je dan iets kon doen in de aard van `if let _ = self as? Authorizeable { insertHeaders(headers)  }` kunt doen?